android/gltfio-android/src/main/cpp/AssetLoader.cpp

```diff
@@ -139,6 +144,49 @@ class JavaMaterialProvider : public MaterialProvider {
         return (MaterialInstance*) mEnv->CallLongMethod(materialInstance, mMaterialInstanceGetNativeObject);
     }
 
+    Material* getMaterial(MaterialKey* config, UvMap* uvmap, const char* label) override {
+        // Create a Java object for the material key and copy the native fields into it.
+        jobject javaKey = mEnv->NewObject(mMaterialKeyClass, mMaterialKeyConstructor);
+
+        auto& helper = MaterialKeyHelper::get();
+        helper.copy(mEnv, javaKey, *config);
+
+        // Convert the optional label into a Java string.
+        jstring stringLabel = label ? mEnv->NewStringUTF(label) : nullptr;
+
+        // Allocate space for the output argument.
+        jintArray uvMapArray = mEnv->NewIntArray(8);
```

Comment:
Where does `8` come from? Should we check that `uvmap->size()` isn't greater?

Reply:
Nice catch, the 8 is fixed but I should not use a magic number here.  Fixed.